LambdaEngine/Include/Input/API/InputActionSystem.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "Input/API/InputState.h"
+#include "Containers/THashTable.h"
+
+namespace LambdaEngine
+{
+	enum EAction {
+		PLAYER_FORWARD,
+		PLAYER_BACKWARD,
+		PLAYER_LEFT,
+		PLAYER_RIGHT,
+		// more actions to come...
+	};
+
+	class LAMBDA_API InputActionSystem
+	{
+	public:
+		DECL_STATIC_CLASS(InputActionSystem);
+
+		// TODO: Read keymapping from file
+		static bool IsActive(EAction action);
+
+		FORCEINLINE static InputActionSystem* GetDevice()
+		{
+			return s_pInputActionSystem;
+		}
```

Comment:
* Would not call this a device, maybe context at most since device refers to some hardware, such as GraphicsDevice, which creates and manages resources on the GPU. AudioDevice manages resources sent to the soundcard, etc. Context stores some form of state. So an InputActionContext would keep the current input state.

Reply:
Yes I was confusing it. I will make a static class instead since no device is needed.